### PR TITLE
Add feature preview landing page examples in Storybook

### PR DESCRIPTION
**Pull request by rezrah**
## Summary

Adding new template examples and deprecating old ones. Also backfilled some missing styles and features required for LPs.

Closes https://github.com/github/primer/issues/2517

Video walkthrough of changes in this PR: https://www.loom.com/share/3043b383685848bfa14e1bfa964758c0

## List of notable changes:

- **added** 3 new templates
- **added** theming controls
- **fixed** failing a11y tests in local environments
- **updated** missing styles for Box
- **added** fullWidth props for SectionIntro and Timeline

## What should reviewers focus on?

- Design review and approval of the landing pages
- Checking story naming and ordering

## Steps to test:

1. Review level 0 in storybook
1. Review level 1 in storybook
1. Review level 2 in storybook

## Supporting resources (related issues, external links, etc):

- [Figma designs](https://www.figma.com/file/m4U5DInCRrbXCwOfWJUfQf/Contentful-Templates?type=design&node-id=0%3A1&mode=design&t=JIM2wKNOOgvcK6oJ-1)

## Contributor checklist:

- [ ] All new and existing CI checks pass
- [ ] Tests prove that the feature works and covers both happy and unhappy paths
- [ ] Any drop in coverage, breaking changes or regressions have been documented above
- [ ] New visual snapshots have been generated / updated for any UI changes
- [ ] All developer debugging and non-functional logging has been removed
- [ ] Related issues have been referenced in the PR description

## Reviewer checklist:

- [ ] Check that pull request and proposed changes adhere to our [contribution guidelines](../../CONTRIBUTING.md) and [code of conduct](../../CODE_OF_CONDUCT.md)
- [ ] Check that tests prove the feature works and covers both happy and unhappy paths
- [ ] Check that there aren't other open Pull Requests for the same update/change

### CODE REVIEW

File: packages/react/src/templates/FeaturePreviewLPs/FeaturePreviewLevelOne/FeaturePreviewLevelOne.tsx [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+                      <Stack direction="vertical" padding="none" alignItems="flex-start">
+                        <Text as="p" variant="muted">
+                          It;s often considered polit to tell people what they are about to sign up for. Even if it
+                          involves making them read a bunch. Lorem ipsum dolor sit amet, consectetur adipiscing elit. In
+                          sapien sit ullamcorper id. Aliquam luctus sed turpis felis nam pulvinar risus elementum.
+                        </Text>
+                        <Link variant="accent" href="#">
+                          Learn more about this
+                        </Link>
+                      </Stack>
```
Comment (danielguillan) [inline packages/react/src/templates/FeaturePreviewLPs/FeaturePreviewLevelOne/FeaturePreviewLevelOne.tsx:97]: Could we use a SectionIntro instead?

File: packages/react/src/templates/FeaturePreviewLPs/FeaturePreviewLevelOne/FeaturePreviewLevelOne.tsx [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+                    </Stack>
+                  </Box>
+                </Grid.Column>
+                <Grid.Column span={6} start={7}>
```
Comment (danielguillan) [inline packages/react/src/templates/FeaturePreviewLPs/FeaturePreviewLevelOne/FeaturePreviewLevelOne.tsx:101]: We should stack this one vertically on smaller viewports

image

File: packages/react/src/templates/FeaturePreviewLPs/FeaturePreviewLevelOne/FeaturePreviewLevelOne.tsx (hunk 1 above)
Comment (rezrah) [reply packages/react/src/templates/FeaturePreviewLPs/FeaturePreviewLevelOne/FeaturePreviewLevelOne.tsx]: Can do 👍, that's updated now.

**Review by danielguillan — APPROVED** [review]

**Review by josepmartins — APPROVED** [review]
LGTM! left a minor comment

**Review by jesussandreas — APPROVED** [review]

**Review by jesussandreas — APPROVED** [review]
no red flags detected, will do another pass with smaller adjustments.

**Pull request merged**